plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js

```diff
@@ -25,6 +28,33 @@ class NodesTableContainer extends mixin(StoreMixin, QueryParamsMixin) {
         suppressUpdate: true
       }
     ];
+
+    this.onMesosStateChange = this.onMesosStateChange.bind(this);
+  }
+
+  componentDidMount() {
+    super.componentDidMount();
```

Comment:
reading through this felt like a anti-pattern so I did a quick research and found that this kinda pattern isn't really suggested.

https://github.com/facebook/react/issues/3280#issuecomment-76383974

Reply:
I know and i felt bad adding this line, but because of the mixin I (though I) had to add it. 
BUT, we were able to remove these functions and instead told the mixin to subscribe 👍